libvast/src/system/spawn_pivoter.cpp

```diff
@@ -31,10 +32,17 @@ spawn_pivoter(node_actor::stateful_pointer<node_state> self,
   // Parse given expression.
   auto query_begin = std::next(arguments.begin());
   expression expr;
-  auto expr_ = system::parse_expression(query_begin, arguments.end());
-  if (!expr_)
-    return expr_.error();
-  expr = *expr_;
+  auto query_result_ = system::parse_query(query_begin, arguments.end());
+  if (!query_result_)
+    return query_result_.error();
+  if (query_result_->second) {
+    return caf::make_error(ec::invalid_configuration,
+                           fmt::format("{} parsed a pipeline string which "
+                                       "should "
+                                       "be impossible",
+                                       *self));
+  }
```

Comment:
Same deal as for the counter.

Reply:
Improved error message in https://github.com/tenzir/vast/pull/2877/commits/d0461071a5c0487d3d189be1978bd17cc56322e9